### PR TITLE
Add a script that replaces all links to records with links to the cor…

### PR DESCRIPTION
**Pull request by jannistsiroyannis**
…responding mainEntities.

### CODE REVIEW

File: whelktool/scripts/cleanups/2020/09/lxl-3152-linkOnlyMainEntity.groovy [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+}
+
+boolean traverse(Object node, String idForLogging, Whelk whelk, PrintWriter linkChanges) {
+    boolean changed = false
+
+    if (node instanceof Map) {
+        Map map = node
+
+        // Correct references are all id.kb.se or ends with #it.
```
Comment (olovy) [inline whelktool/scripts/cleanups/2020/09/lxl-3152-linkOnlyMainEntity.groovy:26]: Code is correct but comment is wrong, could also be an external link (e.g. http://id.loc.gov/authorities/label/Wolves)

**Review by olovy — APPROVED** [review]
LGTM

**Review by olovy — CHANGES_REQUESTED** [review]
Should not change library and script ids
e.g.
https://libris.kb.se/library/NB
https://libris.kb.se/sys/globalchanges/cleanups/2020/01/lxl-2494/repair-uppercase-term-links.groovy

anything else?

File: whelktool/scripts/cleanups/2020/09/lxl-3152-linkOnlyMainEntity.groovy (hunk 1 above)
Comment (jannistsiroyannis) [reply whelktool/scripts/cleanups/2020/09/lxl-3152-linkOnlyMainEntity.groovy]: :+1:

**Comment by jannistsiroyannis** [comment]
> Should not change library and script ids
> e.g.
> https://libris.kb.se/library/NB
> https://libris.kb.se/sys/globalchanges/cleanups/2020/01/lxl-2494/repair-uppercase-term-links.groovy
> 
> anything else?

Good catch, I missed the https://libris.kb.se/library/NB ones. Fixed now. Can't think of anything else!

**Review by klngwll — CHANGES_REQUESTED** [review]
Some other keys, especially so on record, is sensitive in this regards (derivedFrom etc.) I need to dive into this a bit more, but on mainEntity the sameAs is also "special". It is a bit scary with such a broad sweep hehe. I had hoped the problem was isolated to genreForm and perhaps subject.

**Comment by jannistsiroyannis** [comment]
> Some other keys, especially so on record, is sensitive in this regards (derivedFrom etc.) I need to dive into this a bit more, but on mainEntity the sameAs is also "special". It is a bit scary with such a broad sweep hehe. I had hoped the problem was isolated to genreForm and perhaps subject.

It should be noted however, that for any link to a record, where there does not exist any mainEntity (with an id), the links will be left alone. So for example, the heldBy and generationProcess that were given special treatment does not actually need it - they have no mainEntity to link to, so nothing will happen anyway.

**Review by klngwll — APPROVED** [review]
LGTM! 🎈

**Pull request merged**